src/measurements.ts

```diff
@@ -1,80 +1,183 @@
+        this.resetPoints();
       }
     }
   }
 
-  private reset(scene: THREE.Scene) {
-    this.spheres.forEach(sphere => scene.remove(sphere));
-    if (this.line) scene.remove(this.line);
-    this.spheres = [];
-    this.points = [];
-    this.updateMeasurementDisplay("");
-  }
-
   private updateMouse(event: MouseEvent) {
-    this.mouse.x = (event.clientX / window.innerWidth) * 2 - 1;
-    this.mouse.y = -(event.clientY / window.innerHeight) * 2 + 1;
+    this.mouse.set(
+      (event.clientX / window.innerWidth) * 2 - 1,
+      -(event.clientY / window.innerHeight) * 2 + 1
+    );
   }
 
-  private addPoint(scene: THREE.Scene, point: THREE.Vector3) {
+  private addPoint(point: THREE.Vector3) {
     this.points.push(point);
+    const sphere = this.createSphere(point);
+    this.spheresForLine.push(sphere);
 
+    if (this.spheresForLine.length === 2) {
+      this.spheres.push({ sphere1: this.spheresForLine[0], sphere2: this.spheresForLine[1] });
+      this.spheresForLine = [];
+    }
+  }
+
+  private createSphere(point: THREE.Vector3): THREE.Mesh<THREE.SphereGeometry, THREE.MeshBasicMaterial> {
     const sphere = new THREE.Mesh(
       new THREE.SphereGeometry(0.05, 32, 32),
-      new THREE.MeshBasicMaterial({ color: 0xff0000 })
+      new THREE.MeshBasicMaterial({ color: this.color })
     );
     sphere.position.copy(point);
-    scene.add(sphere);
-    this.spheres.push(sphere);
+    this.scene.add(sphere);
+    return sphere;
   }
 
   private calculateDistance() {
     if (this.points.length === 2) {
-      const result = {
-        x: Utils.roundOff(Math.sqrt(this.points[0].x*this.points[0].x + this.points[1].x*this.points[1].x)),
-        y: Utils.roundOff(Math.sqrt(this.points[0].y*this.points[0].y + this.points[1].y*this.points[1].y)),
-        z: Utils.roundOff(Math.sqrt(this.points[0].z*this.points[0].z + this.points[1].z*this.points[1].z)),
-        d: Utils.roundOff(this.points[0].distanceTo(this.points[1]))
-      }
-      this.updateMeasurementDisplay(`Distance = ${JSON.stringify(result)}`);
+      const distance = Utils.roundOff(this.points[0].distanceTo(this.points[1]));
+      this.distances.push(distance);
+      this.updateMeasurementDisplay();
     }
   }
 
-  private drawLine(scene: THREE.Scene) {
+  private drawLine() {
     const geometry = new THREE.BufferGeometry().setFromPoints(this.points);
-    this.line = new THREE.Line(geometry, new THREE.LineBasicMaterial({ color: 0xff0000 }));
-    scene.add(this.line);
+    const line = new THREE.Line(geometry, new THREE.LineBasicMaterial({ color: this.color }));
+    this.colors.push(this.color);
+    this.lines.push(line);
+    this.scene.add(line);
   }
 
-  private updateMeasurementDisplay(text: string) {
-    const measurementContainer = document.getElementById("measurementDiv");
-    if (measurementContainer) measurementContainer.innerText = text;
+  private resetPoints() {
+    this.points = [];
+    this.getRandomColor = true;
```

Comment:
```suggestion
    this.getRandomColor = true;
```

Instead of this new var, we can reset this.color to null